PYME/LMVis/view_clipping_pane.py

```diff
@@ -308,20 +329,41 @@ def OnMouseMove(self, event):
     
         event.Skip()
 
-
-class ViewClippingPanel(wx.Panel):
-    """A GUI class for determining the settings to use when displaying points
-    in VisGUI.
-
-    Constructed as follows:
-    PointSettingsPanel(parent, pipeline, pointDisplaySettings)
-
-    where:
-      parent is the parent window
-      pipeline is the pipeline object which provides the points,
-      pointDisplaySettings is an instance of PointDisplaySettings
+    def ShowTextCtrl(self, text_ctrl, value, xpos):
+        # Update text box position and value
+        ypos = self.Size[1] - text_ctrl.Size[1] + 2
+        text_ctrl.SetPosition((xpos, ypos))
+        text_ctrl.SetValue(str(value))
+
+        # Display
+        text_ctrl.Show()
+        text_ctrl.SetFocus()
+
+    def HideTextCtrl(self, event):
+        event.GetEventObject().Hide()
+
+    def OnTextCtrlKeyPress(self, event):
```

Comment:
Could you bind to `EVT_TEXT_ENTER` instead? (This wouldn't handle escape but would result in less boiler plate). Merely a thought at this point - you don't need to change it.

Reply:
I thought about this. I'm under the impression that `EVT_TEXT_ENTER` would force a change every time the user typed a new number in the box. I thought this would mean handling re-drawing and re-positioning the text box every time the user updated the value. Does it only bind the enter key and/or do I just bind it and skip the re-drawing?

Reply:
As far as I recall, `EVT_TEXT` fires whenever the value changes (i.e. on every character) whereas `EVT_TEXT_ENTER` fires when you press enter or the control looses focus. I might have got the names wrong, but there are definitely 2 separate events with one doing roughly what you'd want.

Reply:
Yes, I just found this in the `wx` docs. I must've been confused or looking at `EVT_TEXT` the last time I looked at this. Thanks for pointing it out.